src/core/importType.js

```diff
@@ -49,6 +49,10 @@ function isRelativeToSibling(name) {
   return name.indexOf('./') === 0
 }
 
+export function isPrivateModule(name) {
+  return name.indexOf('@') === 0
```

Comment:
Ups, renamed to scoped 😄 

Regarding `require('@foo')` as k15a pointed out if you have a folder with that name you'll do `./@foo` and if you have a folder that starts with '@' in your `node_modules` folder well that is the definition a scoped module. If someone create a random folder that follows the same pattern as a scoped package is not unreasonable to expect it to behave like a scoped module. Myself did in the past use scoped modules for local development of multiple packages and nowadays have webpack aliases that use the '@' and load modules from arbitrary locations.